cmd/syncthing/connections.go

```diff
@@ -295,9 +292,7 @@ func (s *connectionSvc) connect() {
 					s.model.Close(deviceID, fmt.Errorf("switching connections"))
 				}
 
-				s.conns <- model.IntermediateConnection{
-					conn, model.ConnectionTypeDirectDial,
-				}
+				s.conns <- relay.Connection{conn, relay.ConnectionTypeDirectDial}
```

Comment:
This is wrong, a dirct connection gets encapsulated as relay connection. Connection package should be separate and have this type.